Update "Getting Set Up to Run REST Assured Tests"

In #9457 we learned that https://guides.dataverse.org/en/5.13/developers/testing.html#getting-set-up-to-run-rest-assured-tests should be updated to at least cover:

- the config added in https://github.com/GlobalDataverseCommunityConsortium/dataverse-ansible/pull/264 to get harvesting tests to pass
- ...

I'm leaving a ... above in case we think of anything else. From a run yesterday of the full API test suite against #9685 as of b627319, the harvesting test above was the only one that failed so that might be it. From https://github.com/IQSS/dataverse/issues/9457#issuecomment-1650561229 it was this test:

 [ERROR] HarvestingServerIT.testMultiRecordOaiSet:606 Wrong number of items on the first ListIdentifiers page expected:<2> but was:<5> 

By the way, there is a detailed note in HarvestingServerIT itself but we should update the guides. Here's that note:

```
// This test will attempt to create a set with multiple records (enough 
// to trigger a paged respons) and test the resumption token functionality). 
// Note that this test requires the OAI service to be configured with some
// non-default settings (the paging limits for ListIdentifiers and ListRecords
// must be set to 2, in order to be able to trigger this paging behavior without
// having to create and export too many datasets).
// So you will need to do this:
//    asadmin create-jvm-options "-Ddataverse.oai.server.maxidentifiers=2"
//    asadmin create-jvm-options "-Ddataverse.oai.server.maxrecords=2"
```

On a related note, I mentioned [in Slack](https://iqss.slack.com/archives/C010LA04BCG/p1690318108213279?thread_ts=1690317077.958369&cid=C010LA04BCG) that I maintain a script called dev-rebuild (see #7363) where I automate these things that developers need to get API tests to pass (createsequence.sql comes to mind). It calls setup-all with --insecure to help developers get set up with the right settings. @landreev and briefly discussed (in that thread) that maybe we need a new flag like --api-tests that can be passed for additional setup such as the harvesting settings and the createsequence thing. So that the API tests can pass. Future work. Out of scope for this issue, which is probably a 3 or our new 0.5.

--- a/doc/sphinx-guides/source/developers/testing.rst
+++ b/doc/sphinx-guides/source/developers/testing.rst
@@ -225,6 +225,20 @@ If ``dataverse.siteUrl`` is absent, you can add it with:
 
 ``./asadmin create-jvm-options "-Ddataverse.siteUrl=http\://localhost\:8080"``
 
+dataverse.oai.server.maxidentifiers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The OAI Harvesting tests require that the paging limit for ListIdentifiers must be set to 2, in order to be able to trigger this paging behavior without having to create and export too many datasets:
+
+``./asadmin create-jvm-options "-Ddataverse.oai.server.maxidentifiers=2"``
+
+dataverse.oai.server.maxrecords
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The OAI Harvesting tests require that the paging limit for ListRecords must be set to 2, in order to be able to trigger this paging behavior without having to create and export too many datasets:
+
+``./asadmin create-jvm-options "-Ddataverse.oai.server.maxrecords=2"``
+
 Identifier Generation
 ^^^^^^^^^^^^^^^^^^^^^
 

